test/pylib/cpp/ldap/prepare_instance.py

```diff
@@ -245,51 +98,83 @@ def try_something_backoff(something):
+        subprocess.check_output(['slaptest', '-f', TOP_SRC_DIR / LDAP_SERVER_CONFIGURATION_FILE, '-F', instance_path],
                                 stderr=subprocess.DEVNULL)
     except:
         pass
     # Set up failure injection.
-    proxy_name = 'p{}'.format(port)
-    subprocess.check_output(
-        ['toxiproxy-cli', 'c', proxy_name, '--listen', 'localhost:{}'.format(port + 2), '--upstream',
-            'localhost:{}'.format(port)])
-    subprocess.check_output(['toxiproxy-cli', 't', 'a', proxy_name, '-t', 'limit_data', '-n', 'limiter', '-a',
-                             'bytes={}'.format(byte_limit)])
-    # Change the data folder in the default config.
-    replace_expression = 's/olcDbDirectory:.*/olcDbDirectory: {}/g'.format(str(instance_path).replace('/', r'\/'))
-    subprocess.check_output(['find', instance_path, '-type', 'f', '-exec', 'sed', '-i', replace_expression, '{}', ';'])
-    # Change the pid file to be kept with the instance.
-    replace_expression = 's/olcPidFile:.*/olcPidFile: {}/g'.format(str(slapd_pid_file).replace('/', r'\/'))
-    subprocess.check_output(['find', instance_path, '-type', 'f', '-exec', 'sed', '-i', replace_expression, '{}', ';'])
-    # Put the test data in.
-    cmd = ['slapadd', '-F', instance_path]
-    subprocess.check_output(cmd, input='\n\n'.join(DEFAULT_ENTRIES).encode('ascii'), stderr=subprocess.STDOUT)
+    try:
+        proxy_name = f'p{port}'
+        subprocess.check_output(
+            ['toxiproxy-cli', '-h', f'{host}:{tp_port}', 'c', proxy_name, '--listen', f'{host}:{port + 2}', '--upstream',
+             f'{host}:{port}'], stderr=subprocess.STDOUT)
+        subprocess.check_output(
+            ['toxiproxy-cli', '-h', f'{host}:{tp_port}', 't', 'a', proxy_name, '-t', 'limit_data', '-n', 'limiter', '-a',
+             f'bytes={toxyproxy_byte_limit}'], stderr=subprocess.STDOUT)
+        # Change the data folder in the default config.
+        replace_expression = f"s/olcDbDirectory:.*/olcDbDirectory: {str(instance_path).replace('/', r'\/')}/g"
+        subprocess.check_output(
+            ['find', instance_path, '-type', 'f', '-exec', 'sed', '-i', replace_expression, '{}', ';'],
+            stderr=subprocess.STDOUT
+        )
+        # Change the pid file to be kept with the instance.
+        replace_expression = f"s/olcPidFile:.*/olcPidFile: {str(slapd_pid_file).replace('/', r'\/')}/g"
+        subprocess.check_output(
+            ['find', instance_path, '-type', 'f', '-exec', 'sed', '-i', replace_expression, '{}', ';'],
+            stderr=subprocess.STDOUT
+        )
+        # Put the test data in.
+        cmd = ['slapadd', '-F', instance_path]
+        subprocess.check_output(cmd, input='\n\n'.join(DEFAULT_ENTRIES).encode('ascii'), stderr=subprocess.STDOUT)
+    except CalledProcessError as e:
+        logging.critical(f"toxiproxy-cli failed: {e}: {e.stdout=}")
```

Comment:
nit, could use placeholders here, so we don't interpolate a string just for formatting the variables. like
```py
logging.critical("toxiproxy-cli failed: %s: %s", e, e.stdout)
```